kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/PodPresetTest.java

```diff
@@ -0,0 +1,208 @@
+      .addNewItem().and()
+      .addNewItem().and().build()).once();
+
+    KubernetesClient client = server.getClient();
+
+    PodPresetList podPresetList = client.settings().podPresets().list();
+    assertNotNull(podPresetList);
+    assertEquals(0, podPresetList.getItems().size());
+
+    podPresetList = client.settings().podPresets().inNamespace("ns1").list();
+    assertNotNull(podPresetList);
+    assertEquals(2, podPresetList.getItems().size());
+  }
+
+  @Test
+  public void testGet(){
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets/podpreset1").andReturn(200, new PodPresetBuilder().build()).once();
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/ns1/podpresets/podpreset2").andReturn(200, new PodPresetBuilder().build()).once();
+
+    KubernetesClient client = server.getClient();
+
+    PodPreset podpreset1 = client.settings().podPresets().withName("podpreset1").get();
+    assertNotNull(podpreset1);
+
+    PodPreset podpreset2 =  client.settings().podPresets().withName("podpreset2").get();
+    assertNull(podpreset2);
+
+    podpreset2 = client.settings().podPresets().inNamespace("ns1").withName("podpreset2").get();
+    assertNotNull(podpreset2);
+  }
+
+  @Test
+  public void testPodUpdate(){
+    KubernetesClient client = server.getClient();
+    try{
+      Pod pod = client.pods().load(getClass().getResourceAsStream("/test-podpreset.yml")).get();
+      assertEquals("pod-test", pod.getMetadata().getName());
+
+      server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets/podpreset-test").andReturn(200, new PodPresetBuilder()
+        .withNewMetadata().withName("podpreset-test").endMetadata()
+        .withNewSpec()
+        .withNewSelector().withMatchLabels(Collections.singletonMap("app", "website")).endSelector()
+        .withEnv(new EnvVarBuilder()
+          .withName("DB_PORT")
+          .withValue("6379")
+          .build())
+        .withVolumeMounts(new VolumeMountBuilder()
+          .withMountPath("/cache")
+          .withName("cache-volume")
+          .build())
+        .withVolumes(
+          new VolumeBuilder()
+            .withName("cache-volume")
+            .withEmptyDir(new io.fabric8.kubernetes.api.model
+              .EmptyDirVolumeSourceBuilder()
+              .build())
+            .build())
+        .endSpec()
+        .build())
+        .once();
+
+      pod = client.pods().inNamespace("default").withName(pod.getMetadata().getName()).get();
+      assertEquals("pod-test", pod.getMetadata().getName());
+      assertNotNull(pod.getMetadata().getAdditionalProperties().get("DB_PORT"));
+      assertNotNull(pod.getMetadata().getAdditionalProperties().get("cache-volume"));
+
+      PodPreset podpreset = client.settings().podPresets().withName("podpreset-test").get();
+      assertEquals("podpreset-test", podpreset.getMetadata().getName());
+    } catch (Exception e) {
```

Comment:
You should fix this NullPointerException:
```
INFO: --> END GET
Jun 10, 2019 1:18:59 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[60617] received request: GET /api/v1/namespaces/default/pods/pod-test HTTP/1.1 and responded: HTTP/1.1 404 Client Error
Jun 10, 2019 1:18:59 PM okhttp3.internal.platform.Platform log
INFO: <-- 404 Client Error https://localhost:60617/api/v1/namespaces/default/pods/pod-test (15ms)
Jun 10, 2019 1:18:59 PM okhttp3.internal.platform.Platform log
INFO: Content-Length: 0
Jun 10, 2019 1:18:59 PM okhttp3.internal.platform.Platform log
INFO: <-- END HTTP (0-byte body)
java.lang.NullPointerException
	at io.fabric8.kubernetes.client.mock.PodPresetTest.testPodUpdate(PodPresetTest.java:107)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Jun 10, 2019 1:22:17 PM okhttp3.internal.platform.Platform log
INFO: --> DELETE https://localhost:60617/api/v1/namespaces/test/pods/pod-test http/1.1
Jun 10, 2019 1:22:17 PM okhttp3.internal.platform.Platform log
INFO: Content-Type: application/json; charset=utf-8
Jun 10, 2019 1:22:17 PM okhttp3.internal.platform.Platform log
INFO: Content-Length: 89
Jun 10, 2019 1:22:17 PM okhttp3.internal.platform.Platform log
```

Reply:
Fixed